packages/vm/src/testing/contract.rs

```diff
@@ -95,15 +99,30 @@ impl<'a> Contract<'a> {
     pub fn change_options(&mut self, options: MockInstanceOptions<'a>) {
         self.options = options
     }
+
+    /// update balance in backend querier. it does not change the options
+    pub fn update_balance<T: Into<String>>(&mut self, addr: T, balance: Vec<Coin>) -> TestingResult<Option<Vec<Coin>>> {
+        let mut backend = self
+            .backend
+            .take()
+            .ok_or_else(|| TestingError::ContractError(ERR_UPDATE_BALANCE_OF_INSTANCE.to_string()))?;
+
+        // update balances of querier
+        let res = backend.querier.update_balance(addr, balance);
+        self.backend = Some(backend);
```

Comment:
please see the above opinion. I think if we not use the `take`,  we don't need this line.

Reply:
ditto.